README.md

```diff
@@ -10,14 +10,158 @@
+
+![alt text](docs/setup/resources/hubspot-new-app-auth.png)
+
+![alt text](docs/setup/resources/hubspot-new-app-redirect-url.png)
+
+### Setp 2. Obtain Client ID and Client Secret.
+
+1. After completing the project setup, you will be provided with your client id and client secret.  Make sure to save the provided client id and client secret.
+
+![alt text](docs/setup/resources/hubspot-new-app-client-id-secret.png)
+
+### 2. Setup OAuth 2.0 Flow
+
+Before proceeding with Quickstart, ensure you have obtained the Access Token and refresh Token using the following steps:
+
+1. Add necessary scopes for your app based on API your using. Go to the relevate [API reference](https://developers.hubspot.com/beta-docs/reference/api),  select the API  you have  and go through the operation.
+
+You will see the scope has defined below way
+
+![alt text](docs/setup/resources/hubspot-new-app-scopes.png)
+
+2. Obtained the authorization URL (Install URL) from the Auth Section under 'Sample install URL (OAuth)' section.  It will be in this format:
+
+```
+https://app.hubspot.com/oauth/authorize?client_id=<client_id>&redirect_uri=<redirect_url>&scope=<scopes>
+```
+3. Copy and paste the generated URL into your browser. This will redirect you to the HubSpot authorization page.
+
+![alt text](docs/setup/resources/hubspot-oauth-consent-screen.png)
+
+4. Once you authorize, you will be redirected to your specified redirect URI with an authorization code in the URL.
+
+**Note**: Store the authorization code and use it promptly as it expires quickly.
+
+5. Use the obtained authorization code to run the following curl command, replacing <your_client_id>, <your_redirect_url>, and <your_authorization_code> with your specific values:
+
+ - Linux/MacOS:
+```bash
+curl --request POST \
+  --url https://api.hubapi.com/oauth/v1/token \
+  --header 'content-type: application/x-www-form-urlencoded' \
+  --data 'grant_type=authorization_code&code=<your_authorization_code>&redirect_uri=<your_redirect_url>&client_id=<your_client_id>&client_secret=<your_client_secret>'
+```
+
+ - Windows:
+ ```bash
+ curl --request POST \
+  --url https://api.hubapi.com/oauth/v1/token ^
+  --header 'content-type: application/x-www-form-urlencoded' ^
+  --data 'grant_type=authorization_code&code=<your_authorization_code>&redirect_uri=<your_redirect_url>&client_id=<your_client_id>&client_secret=<your_client_secret>'
+ ```
+
+ This command will return the access token and refresh token necessary for API calls.
+
+ ```json
+{
+  "token_type": "bearer",
+  "refresh_token": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+  "access_token": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+  "expires_in": 1800
+}
+ ```
+
+ 6. Store the access token securely for use in your application.
+
+
 ## Quickstart
 
 [//]: # (TODO: Add a quickstart guide to demonstrate a basic functionality of the module, including sample code snippets.)
 
```

Comment:
No need for this text as we have already added the relevant details. Please check for other places as well.
```suggestion
```